### PR TITLE
feat(api,ui): first iteration on worker model view

### PR DESCRIPTION
**Pull request by sguiheux**
Add ascode editor with yaml key autocompletion

### CODE REVIEW

File: engine/api/v2_entities.go [hunk 1]
```diff
@@ -77,3 +77,61 @@ func (api *API) getEntitiesHandler() ([]service.RbacChecker, service.Handler) {
...
+				return err
+			}
+
+			repo, err := api.getRepositoryByIdentifier(ctx, vcsProject.ID, repositoryIdentifier)
+			if err != nil {
+				return err
+			}
+
+			if branch == "" {
+				tx, err := api.mustDB().Begin()
```
Comment (fsamin) [inline engine/api/v2_entities.go:110]: why do you need a transaction to get the default branch ?

File: engine/api/v2_entities.go [hunk 2]
```diff
@@ -77,3 +77,61 @@ func (api *API) getEntitiesHandler() ([]service.RbacChecker, service.Handler) {
 			return service.WriteJSON(w, result, http.StatusOK)
 		}
 }
+
+func (api *API) getEntityHandler() ([]service.RbacChecker, service.Handler) {
```
Comment (fsamin) [inline engine/api/v2_entities.go:81]: i think you forget to push the unit test file :)

**Review by fsamin — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18336.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

File: engine/api/v2_entities.go (hunk 1 above)
Comment (sguiheux) [reply engine/api/v2_entities.go:110]: VCS Client need a transaction since we implement bitbucketcloud

**Review by fsamin — APPROVED** [review]

**Review by yesnault — APPROVED** [review]

**Review by richardlt — APPROVED** [review]

**Pull request merged**